lib/livebook/file_system/s3.ex

```diff
@@ -30,8 +30,11 @@ defmodule Livebook.FileSystem.S3 do
     bucket_url = String.trim_trailing(bucket_url, "/")
     region = opts[:region] || region_from_uri(bucket_url)
 
+    hash = :crypto.hash(:sha256, bucket_url) |> Base.url_encode64(padding: false)
+    id = "s3-#{hash}"
```

Comment:
Currently we are using random ids generated when the file system is added, which is fine for settings (local). However, to persist a file link in a notebook, we need a deterministic file system id, so if multiple Livebook users use the same S3 bucket, the file link is exported/imported as expected.

I did the hash so that the id is fixed length and this way it also has a predictable character set. Also, even though buckets generally need authorisation, not listing the URL explicitly sounds like an advantage to me.

I considered a second field/function like `:public_id`, but having both random `:id` and the `:public_id` is confusing, especially that both should be unique anyway. To avoid breaking existing settings I added a migration with more details.

Reply:
Looks good to me. I also wonder if the ID should be the bucket itself but this at least avoids us from leaking bucket information.

Reply:
Yeah, I would still like base64 encode the bucket URL to avoid weird characters in the ID, but not leaking sounds preferable anyway.